src/api-umbrella/proxy/jobs/distributed_rate_limit_puller.lua

```diff
@@ -89,7 +78,10 @@ local function check(premature)
     return
   end
 
-  local ok, err = pcall(do_check)
+  -- here we subtract the lock expiration time by 1ms to prevent
+  -- a race condition with the next timer event.
+  local ok, err = interval_lock('distributed-last-pull', delay - 0.001,
```

Comment:
Whoopsie; last minute edit fail. The intent was originally the reverse -- why would `interval_lock` know about a timer? It's a safe play, but I think it assumes too much knowledge about it's calling context.

Given that we're expecting to combine this sort of lock with a timer often, perhaps it makes more sense to add a related function which also sets a timer for callbacks?